src/main/kotlin/de/cyface/uploader/DefaultUploader.kt

```diff
@@ -66,54 +68,53 @@ import javax.net.ssl.SSLException
  * To use this interface just call [DefaultUploader.uploadMeasurement] or [DefaultUploader.uploadAttachment].
  *
  * @author Armin Schnabel
- * @version 2.0.0
- * @since 1.0.0
  * @property apiEndpoint An API endpoint running a Cyface data collector service, like `https://some.url/api/v3`
  */
 class DefaultUploader(private val apiEndpoint: String) : Uploader {
 
     @Suppress("unused", "CyclomaticComplexMethod", "LongMethod") // Part of the API
     override fun uploadMeasurement(
         jwtToken: String,
-        metaData: RequestMetaData,
+        uploadable: Measurement,
         file: File,
         progressListener: UploadProgressListener
     ): Result {
         val endpoint = measurementsEndpoint()
-        return uploadFile(jwtToken, metaData, file, endpoint, progressListener)
+        return uploadFile(jwtToken, uploadable, file, endpoint, progressListener)
     }
 
     override fun uploadAttachment(
         jwtToken: String,
-        metaData: RequestMetaData,
-        measurementId: Long,
+        uploadable: Attachment,
         file: File,
         fileName: String,
-        progressListener: UploadProgressListener
+        progressListener: UploadProgressListener,
     ): Result {
-        val endpoint = attachmentsEndpoint(measurementId)
-        return uploadFile(jwtToken, metaData, file, endpoint, progressListener)
+        val measurementId = uploadable.identifier.measurementIdentifier
```

Comment:
I want to keep the PRs as small as possible, so I didn't try to refactor further here.